src/cargo/util/toml/mod.rs

```diff
@@ -2151,6 +2164,40 @@ fn to_dependency_source_id<P: ResolveToPath + Clone>(
     }
 }
 
+pub(crate) fn lookup_path_base(
+    base: &str,
+    gctx: &GlobalContext,
+    manifest_root: &Path,
+) -> CargoResult<PathBuf> {
+    if !gctx.cli_unstable().path_bases {
+        bail!("usage of path bases requires `-Z path-bases`");
+    }
+
+    // Look up the relevant base in the Config and use that as the root.
+    if let Some(path_bases) =
+        gctx.get::<Option<ConfigRelativePath>>(&format!("path-bases.{base}"))?
```

Comment:
Aside: I wonder how this is on performance and if we should load all of `path-bases` and then index into it (that also removes tracking of invariants to prevent path-injection)

Reply:
I guess it depends on whether we think there will be more items in the config.toml (i.e., making the index is expensive) or more uses in the Cargo.toml (i.e., having to look up the same items many times). For now, I'd rather wait and see how the feature is used, but I definitely add this as a note in the tracking issue.

Reply:
If its in the tracking issue, we can cover it later

Reply:
ok